### PR TITLE
Weapon Components for Spawning Projectiles

### PR DESCRIPTION
**Pull request by cdsupina**
- added weapon component for spawning projectiles
- move values to describe projectile spawning into weapon component
- projectiles now spawn rotated in the direction they were spawned
- fix fire rate scaling with money to work with both characters and account for base reload time of character
- nerf the megablast ability to fire multiplier/2 blasts

### CODE REVIEW

File: src/game/counters/plugin.rs [hunk 1]
```diff
@@ -293,6 +307,7 @@ mod test {
             .get(&DEFAULT_USER_ID)
             .unwrap()
             .total_shots_fired;
+        println!("n1_p1_shots_fired_2: {n_p1_shots_fired_2}");
```
Comment (varoonp123) [inline src/game/counters/plugin.rs]:
```suggestion

```

File: src/run/tutorial.rs [hunk 2]
```diff
@@ -305,18 +308,18 @@ impl TutorialLesson {
...
                 Self::Ability { .. } => {
-                    player.disable_main_attacks();
+                    weapon.disable();
                 }
                 Self::Attack { .. } => {
                     player.disable_special_attacks();
                 }
                 Self::Movement { .. } => {
-                    player.disable_main_attacks();
+                    weapon.disable();
```
Comment (varoonp123) [inline src/run/tutorial.rs:322]: One `disable` is on `player: PlayerComponent` and one `disable` in on `WeaponComponent`

**Comment by varoonp123** [comment]
The one thing I would like before merging this is resolving the question "can a player/mob have more than 1 weapon/attack?" rather than handling them separately. What if we want to let you pick up ~5 attacks (a freeze, mind control, chain/lightning etc.) that could be toggled or all selected (like those games where you press 1-9 to take actions)? 

Maybe you can only have 1 weapon but that weapon can have separate attacks? Idk

File: src/run/tutorial.rs (hunk 2 above)
Comment (cdsupina) [reply src/run/tutorial.rs:322]: This is how it has to be at this point. I plan on overhauling abilities next. When this is done, the ability will be reclassified as another weapon that can be controlled independently from the player component.

**Review by varoonp123 — APPROVED** [review]

**Pull request merged**